testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/observability/micrometer/MicrometerSetupTask.java

```diff
@@ -5,118 +5,68 @@
-            startOpenTelemetryCollector();
-        } catch (Exception e) {
-            System.err.println("OpenTelemetry Container failed to start.");
-        }
-
-        executeOp(managementClient, enableStatistics(true));
+        executeOp(managementClient, writeAttribute("undertow", STATISTICS_ENABLED, "true"));
 
-        ModelNode outcome = executeRead(managementClient, micrometerExtension);
-        if (!Operations.isSuccessfulOutcome(outcome)) {
+        if (!Operations.isSuccessfulOutcome(executeRead(managementClient, micrometerExtension))) {
             executeOp(managementClient, Operations.createAddOperation(micrometerExtension));
             extensionAdded = true;
         }
 
-        outcome = executeRead(managementClient, micrometerSubsystem);
-        if (!Operations.isSuccessfulOutcome(outcome)) {
+        if (!Operations.isSuccessfulOutcome(executeRead(managementClient, micrometerSubsystem))) {
             ModelNode addOp = Operations.createAddOperation(micrometerSubsystem);
             addOp.get("endpoint").set("http://localhost:4318/v1/metrics"); // Default endpoint
             executeOp(managementClient, addOp);
             subsystemAdded = true;
         }
 
-        if (containerStarted) {
-            executeOp(managementClient, writeAttribute("micrometer", "endpoint", otelCollector.getOtlpEndpoint()));
+        if (dockerAvailable) {
+            otelCollector = OpenTelemetryCollectorContainer.getInstance();
+            executeOp(managementClient, writeAttribute("micrometer", "endpoint",
+                    otelCollector.getOtlpHttpEndpoint() + "/v1/metrics"));
+            executeOp(managementClient, writeAttribute("micrometer", "step", "1"));
         }
-        executeOp(managementClient, writeAttribute("micrometer", "step", "1"));
 
         ServerReload.executeReloadAndWaitForCompletion(managementClient);
     }
 
     @Override
     public void tearDown(final ManagementClient managementClient, String containerId) throws Exception {
-        if (containerStarted) {
+        if (dockerAvailable) {
             otelCollector.stop();
-            otelCollector = null;
-            containerStarted = false;
-        }
-
-        executeOp(managementClient, enableStatistics(false));
-        if (subsystemAdded) {
-            executeOp(managementClient, Operations.createRemoveOperation(micrometerSubsystem));
-        }
-        if (extensionAdded) {
-            executeOp(managementClient, Operations.createRemoveOperation(micrometerExtension));
-        }
 
-        ServerReload.reloadIfRequired(managementClient);
-    }
+            executeOp(managementClient, clearAttribute("undertow", STATISTICS_ENABLED));
+            if (subsystemAdded) {
+                executeOp(managementClient, Operations.createRemoveOperation(micrometerSubsystem));
+            }
+            if (extensionAdded) {
+                executeOp(managementClient, Operations.createRemoveOperation(micrometerExtension));
+            }
```

Comment:
L61-67 should be outside the 'if (dockerAvailable) block, since they are outside of it in setup.

Reply:
Good catch. Fixed.